gourmet/tests/test_plugin_loader.py

```diff
@@ -1,37 +1,50 @@
 import unittest
 import tempfile
 import os
+import shutil
 
-import gourmet.gglobals
+import gourmet.gglobals as gglobals
 
-# clear out Gourmet's DB
-tmpdir = tempfile.mktemp()
-os.makedirs(tmpdir)
-gourmet.gglobals.gourmetdir = tmpdir
-
-import gourmet.GourmetRecipeManager
-import gourmet.backends.db
-
-gourmet.backends.db.RecData.__single = None
-gourmet.GourmetRecipeManager.GourmetApplication.__single = None
-# end clearing out code
-
-from gourmet.plugin_loader import get_master_loader
 
 class Test (unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # create a temporary directory for tests
+        cls.tmp_dir = tempfile.mktemp(".pl")  # TODO: replace deprecated mktemp()
+        os.makedirs(cls.tmp_dir)
+        gglobals.gourmetdir = cls.tmp_dir
+
+        # continue to import with gourmetdir location set to tmp_dir
+        from gourmet.GourmetRecipeManager import GourmetApplication
+        from gourmet.backends import db
+        from gourmet.plugin_loader import get_master_loader
+
+        db.RecData.__single = None
+        GourmetApplication.__single = None
+        cls.ml = get_master_loader()
+
+    @classmethod
+    def tearDownClass(cls):
+        # delete temporary test database
+        try:
+            shutil.rmtree(cls.tmp_dir)
+        except OSError as e:
+            print("Error: {} : {}").format(cls.tmp_dir, e.strerror)
 
     def testDefaultPlugins (self):
-        ml = get_master_loader()
-        ml.load_active_plugins()
-        print('active:',ml.active_plugins)
-        print('instantiated:',ml.instantiated_plugins)
-        assert(not ml.errors)
+        self.ml.load_active_plugins()
+        print('active:',self.ml.active_plugins)
+        print('instantiated:',self.ml.instantiated_plugins)
+        self.assertEqual(len(self.ml.errors), 0)  # there should be 0 plugin errors
 
     def testAvailablePlugins (self):
-        ml = get_master_loader()
-        for st in ml.available_plugin_sets:
-            if st not in ml.active_plugins:
-                ml.activate_plugin_set(st)
+        # search module directories for available plugins
+        for module_name, plugin_set in self.ml.available_plugin_sets.items():
+            if module_name not in self.ml.active_plugins:
```

Comment:
I think this needs to check `in self.ml.active_plugin_sets` - see also #124.

I figured out that a 'plugin set' refers to a Python module which may have several 'plugin' classes to extend different bits of the application (database plugin, preferences plugin, etc.). The `nutritional_information` module is one example that has several different plugins in one plugin set.

Reply:
As suggested, I've pulled in your changes from your branch `testAvailablePlugins` into my branch. I will update the PR with all the changes recommend.